src/blueapi/service/interface.py

```diff
@@ -51,27 +51,26 @@ def worker() -> TaskWorker:
 @lru_cache
 def stomp_client() -> StompClient | None:
     stomp_config = config().stomp
-    if stomp_config is not None:
-        template = StompClient.for_broker(
-            broker=Broker(
-                host=stomp_config.host, port=stomp_config.port, auth=stomp_config.auth
-            )
+    if stomp_config is None:
+        return None
```

Comment:
Should: Please revert this approach

I'm picking this out as an example but there are many others in this PR where you achieve exhaustiveness via early returns rather than explicit flow control i.e.

```diff
def foo() -> int:
    if bar:
        return 2
-    else:
-        return 5 
+   return 5
```

This is quite an opinionated change and contrary to the approach used so far in the codebase. In reality the above diff makes no functional change and, in my view, is just less readable because it removes context and connecting words.

I imagine there's a fair bit of literature about this arguing for different patterns and paradigms but it would be a bit bike sheddy to go through it all. I would suggest just trying to be consistent with the codebase as you found it. If you really find the pattern that's there unreadable, please explain why and we can try to figure out how to make things better.

Reply:
>  you achieve exhaustiveness via early returns rather than explicit flow control 

yes this is on purpose

but ok if this is in the codebase it makes no sense to fight it


but
> In reality the above diff makes no functional change 

that's the defintion of refactoring

> connecting words

different logical branches are decoupled, they don't need connection

Reply:
> reads better than

that's subjective


'if-else' assumes it's 2 branches. 'early return' reads the same when the number of branches rises, and is more appropriate than a switch statement in cases when the branches aren't very _cleanly_ different based on `typeof someVariable`.

`if-else', if more branches grow becomes a hard structure to track.

furthermore

> translates more directly into natural language

it's not obvious whether that is good or bad...


https://www.youtube.com/watch?v=EumXak7TyQ0

https://developer20.com/elseless/

Reply:
I clearly didn't state my argument well enough. 

trying again ...

for any location in code where _business logic_ has multiple branches, that must be reflected in the code. 

patterns are many: switch statements, wrapping callbacks in a dictionary (like for filter_by_status), if + early return, `if-else`. 

Rust uses switch-y pattern matching very often. 

in Python when we have 10 branches we'll almost always go for switch, which would feel cumbersome if there was just 2.

early return, which is my favourite isn't as great if there are 10 branches, visually, scales better than `if-else` 

with early return when I read a function I keep track of branches and possible end-states of the function. 
Every if with early return opens one possible end state and then closes it in the same place. 

```python

if a:
  do_x()
  return
if b:
  do_y()
do_z()
```
with `if-else` it's easier to get confused as each layer cuts in 2 the possibility space and as a reader I now need to track 2^n possibilities and see if the writer accounted correctly for all of them. 

therefore the 'as the number of branches increases' refers to both lifetime of the codebase and the distribution of business logic instances in complexity from the start.

early return is more futureproof and a habit to avoid making if-else where some of the possiblities aren't addressed